android/runtime/v8/src/native/modules/APIModule.cpp

```diff
@@ -115,7 +115,13 @@ void APIModule::logInfo(const FunctionCallbackInfo<Value>& args)
 	Isolate* isolate = args.GetIsolate();
 	HandleScope scope(isolate);
 	v8::String::Utf8Value message(isolate, APIModule::combineLogMessages(args));
-	APIModule::logInternal(LOG_LEVEL_INFO, LCAT, *message);
+
+	std::string cppStr(*message);
+
+	int maxChunk = 4050;
+	for (int i = 0; i < cppStr.length(); i+=maxChunk) {
```

Comment:
`i` is a signed `int` but `.length()` returns a `size_t` which is an `unsigned int`. There's a loss of data when the value is cast. To prevent the loss of data and speed up the loop by only evaluating the length once, move the length to the for loop initialization variables and use `size_t`.

```suggestion
	for (size_t i = 0, len = cppStr.length(); i < len; i+=maxChunk) {
```

Reply:
runs fine, thanks :+1: